src/containers/pages/InterventionPlan/UpdatePlan/PlanLocationNames/index.tsx

```diff
@@ -0,0 +1,115 @@
+/** will be responsible for lazy loading location names that are
+ * assigned to a specific plan
+ */
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import React, { ReactElement, useEffect } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { Store } from 'redux';
+import Loading from '../../../../../components/page/Loading';
+import { PlanDefinition } from '../../../../../configs/settings';
+import { FI_SINGLE_URL, OPENSRP_LOCATIONS_BY_PLAN } from '../../../../../constants';
+import { displayError } from '../../../../../helpers/errors';
+import { OpenSRPService } from '../../../../../services/opensrp';
+import store from '../../../../../store';
+import locationsReducer, {
+  fetchLocations,
+  getLocationsByPlanId,
+  Location,
+  reducerName as locationsReducerName,
+} from '../../../../../store/ducks/opensrp/locations';
+
+// register the locations reducer.
+reducerRegistry.register(locationsReducerName, locationsReducer);
+
+/** interface for props for the JurisdictionIdsToNames component */
+interface Props {
+  child?: (locationName: string, locationId: string, index: number) => ReactElement;
+  fetchLocationsAction: typeof fetchLocations;
+  locations: Location[];
+  plan: PlanDefinition | null;
+  serviceClass: typeof OpenSRPService;
+}
+
+const defaultProps: Props = {
+  fetchLocationsAction: fetchLocations,
+  locations: [],
+  plan: null,
+  serviceClass: OpenSRPService,
+};
+
+/** The PlanLocationNames component takes a plan and uses its id
+ * to get location names which it renders in a list
+ */
+const PlanLocationNames = (props: Props) => {
+  const { serviceClass, fetchLocationsAction, plan } = props;
+
+  /** async function that takes in a planId and calls
+   * performs the api call and then updates the store
+   *
+   * @param {typeof OpenSRPService} service - the opensrp service class
+   * @param {string} planId - fetch locations attached to plan with this id
+   * @param {typeof fetchLocations} actionCreator - action creator
+   */
+  async function loadLocations(
+    service: typeof OpenSRPService,
+    planId: string,
+    actionCreator: typeof fetchLocations
+  ) {
+    // perform api call and dispatch result
+    const serve = new service(OPENSRP_LOCATIONS_BY_PLAN);
+
+    serve
+      .read(planId)
+      .then((response: Location[]) => {
+        store.dispatch(actionCreator(response, planId));
+      })
+      .catch((err: Error) => {
+        displayError(err);
+      });
+  }
```

Comment:
@p-netm would it be possible to move this out of the component and export it as its own const? (for testability?)

Reply:
done